src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseDataFileService.cs

```diff
@@ -392,8 +445,13 @@ private async Task<List<DataFileInfo>> BuildDataFileViewModels(List<ReleaseFile>
             };
         }
 
-        private async Task<string> GetSubjectName(Guid releaseId, File file)
+        private async Task<string> GetSubjectName(Guid releaseId, File? file)
         {
+            if (file == null)
+            {
+                throw new ArgumentException("file should not be null");
+            }
+
```

Comment:
Similar to earlier - this seems like a big code smell as we accept an optional parameter that immediately throws if it's not present.

We should revert this change and handle any null `file` parameter in the consuming method i.e. `ValidateSubjectName`.